<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>843</width>
    <height>648</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabWorldmap">
       <attribute name="title">
        <string>Worldmap Editor</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="WorldmapEditor" name="worldmapEditor" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabLevelDesigner">
       <attribute name="title">
        <string>Level Designer</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="1" column="1" colspan="2">
         <widget class="QProgressBar" name="progressBar">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="value">
           <number>100</number>
          </property>
          <property name="textVisible">
           <bool>true</bool>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="invertedAppearance">
           <bool>false</bool>
          </property>
          <property name="format">
           <string>%p%</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="listMap"/>
        </item>
        <item row="4" column="2">
         <widget class="QWidget" name="widget_2" native="true">
          <layout class="QGridLayout" name="gridLayout_16">
           <item row="1" column="0">
            <widget class="QPushButton" name="mapMoveLeft">
             <property name="text">
              <string>←</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="mapMoveBottom">
             <property name="text">
              <string>↓</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="mapMoveRight">
             <property name="text">
              <string>→</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="mapMoveTop">
             <property name="text">
              <string>↑</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QCheckBox" name="freeCamera">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="text">
              <string>Free Camera</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QCheckBox" name="lockCamera">
             <property name="text">
              <string>Lock camera</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QWidget" name="widget_4" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_10">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QTabWidget" name="map_tabs">
             <property name="currentIndex">
              <number>0</number>
             </property>
             <widget class="QWidget" name="map_tree">
              <attribute name="title">
               <string>Object Tree</string>
              </attribute>
              <layout class="QGridLayout" name="gridLayout_12">
               <item row="1" column="0">
                <widget class="QPushButton" name="objectAddButton">
                 <property name="text">
                  <string>Add</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QPushButton" name="objectDeleteButton">
                 <property name="text">
                  <string>Delete</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="0" colspan="2">
                <widget class="WorldObjectWidget" name="worldObjectWidget" native="true"/>
               </item>
               <item row="0" column="0" colspan="2">
                <widget class="MapTreeWidget" name="treeWidget">
                 <property name="dragEnabled">
                  <bool>true</bool>
                 </property>
                 <property name="dragDropMode">
                  <enum>QAbstractItemView::DragDrop</enum>
                 </property>
                 <property name="sortingEnabled">
                  <bool>true</bool>
                 </property>
                 <property name="animated">
                  <bool>true</bool>
                 </property>
                 <property name="headerHidden">
                  <bool>true</bool>
                 </property>
                 <column>
                  <property name="text">
                   <string notr="true">1</string>
                  </property>
                 </column>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="terrain">
              <attribute name="title">
               <string>Terrain</string>
              </attribute>
              <layout class="QGridLayout" name="gridLayout_9">
               <item row="1" column="1">
                <widget class="QPushButton" name="waypointDiscardSelection">
                 <property name="text">
                  <string>Discard Selection</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QPushButton" name="waypointPicker">
                 <property name="text">
                  <string>Picker</string>
                 </property>
                 <property name="checkable">
                  <bool>true</bool>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="6" column="0">
                <widget class="Line" name="line">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QPushButton" name="waypointZoneSelector">
                 <property name="text">
                  <string>Zone selection</string>
                 </property>
                 <property name="checkable">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="0" column="0" colspan="2">
                <widget class="QPushButton" name="waypointGenerateObject">
                 <property name="text">
                  <string>Generate for current object</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QCheckBox" name="waypointVisible">
                 <property name="text">
                  <string>Visible</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="3" column="0" colspan="2">
                <widget class="QFrame" name="waypointEditor">
                 <property name="frameShape">
                  <enum>QFrame::StyledPanel</enum>
                 </property>
                 <property name="frameShadow">
                  <enum>QFrame::Raised</enum>
                 </property>
                 <layout class="QGridLayout" name="gridLayout_8">
                  <item row="9" column="0">
                   <widget class="QLabel" name="label">
                    <property name="text">
                     <string>Move</string>
                    </property>
                   </widget>
                  </item>
                  <item row="11" column="0" colspan="3">
                   <widget class="QPushButton" name="waypointClone">
                    <property name="text">
                     <string>Clone last selected waypoint</string>
                    </property>
                    <property name="flat">
                     <bool>false</bool>
                    </property>
                   </widget>
                  </item>
                  <item row="3" column="0" colspan="3">
                   <widget class="QLineEdit" name="waypointSelection">
                    <property name="enabled">
                     <bool>false</bool>
                    </property>
                   </widget>
                  </item>
                  <item row="5" column="0" colspan="3">
                   <widget class="QPushButton" name="waypointDisconnect">
                    <property name="text">
                     <string>Disconnect</string>
                    </property>
                    <property name="flat">
                     <bool>false</bool>
                    </property>
                   </widget>
                  </item>
                  <item row="10" column="2">
                   <widget class="QDoubleSpinBox" name="waypointSelZ">
                    <property name="minimum">
                     <double>-999.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>999.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="10" column="0">
                   <widget class="QDoubleSpinBox" name="waypointSelX">
                    <property name="minimum">
                     <double>-999.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>999.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>0.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="7" column="0" colspan="3">
                   <widget class="QPushButton" name="waypointSelDelete">
                    <property name="text">
                     <string>Remove</string>
                    </property>
                    <property name="flat">
                     <bool>false</bool>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_2">
                    <property name="text">
                     <string>&lt;b&gt;Selection&lt;/b&gt;</string>
                    </property>
                   </widget>
                  </item>
                  <item row="4" column="0" colspan="3">
                   <widget class="QPushButton" name="waypointConnect">
                    <property name="text">
                     <string>Connect</string>
                    </property>
                    <property name="flat">
                     <bool>false</bool>
                    </property>
                   </widget>
                  </item>
                  <item row="10" column="1">
                   <widget class="QDoubleSpinBox" name="waypointSelY">
                    <property name="minimum">
                     <double>-999.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>999.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item row="12" column="0" colspan="3">
                   <widget class="QPushButton" name="forceFloorAbove">
                    <property name="text">
                     <string>Force floor above</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="zonesTab">
              <attribute name="title">
               <string>Zones</string>
              </attribute>
              <layout class="QGridLayout" name="gridLayout_6">
               <item row="3" column="0">
                <widget class="QLabel" name="zoneLabelDestination">
                 <property name="text">
                  <string>Destinations:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0" colspan="2">
                <widget class="QPushButton" name="zoneAdd">
                 <property name="text">
                  <string>Add</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QComboBox" name="zoneList"/>
               </item>
               <item row="1" column="1">
                <widget class="QPushButton" name="zoneDelete">
                 <property name="text">
                  <string>Delete</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0" colspan="2">
                <widget class="QPushButton" name="zoneSetSelection">
                 <property name="text">
                  <string>Set as selection</string>
                 </property>
                </widget>
               </item>
               <item row="5" column="0" rowspan="2" colspan="2">
                <widget class="QListWidget" name="zoneDestinations"/>
               </item>
               <item row="7" column="0">
                <widget class="QPushButton" name="zoneAddDestination">
                 <property name="text">
                  <string>Add</string>
                 </property>
                </widget>
               </item>
               <item row="7" column="1">
                <widget class="QPushButton" name="zoneDeleteDestination">
                 <property name="text">
                  <string>Delete</string>
                 </property>
                </widget>
               </item>
               <item row="8" column="0">
                <widget class="Line" name="line_2">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="1" rowspan="10">
         <widget class="QPandaWidget" name="widget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <layout class="QGridLayout" name="gridLayout"/>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QWidget" name="widget_3" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QPushButton" name="mapNew">
             <property name="text">
              <string>New Map</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="saveMap">
             <property name="text">
              <string>Save Map</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabStatsheets">
       <attribute name="title">
        <string>Statsheets</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_15">
        <item row="0" column="0">
         <widget class="QPushButton" name="charsheetAdd">
          <property name="text">
           <string>New</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="charsheetDel">
          <property name="text">
           <string>Delete</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QListWidget" name="charsheetList">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
         </widget>
        </item>
        <item row="3" column="2" rowspan="2" colspan="3">
         <widget class="CharsheetEditor" name="charsheetEditor" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QPushButton" name="charsheetSave">
          <property name="text">
           <string>Save</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabScriptEditor">
       <attribute name="title">
        <string>Script Editor</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="2" column="0" colspan="2">
         <widget class="QTreeWidget" name="scriptList">
          <column>
           <property name="text">
            <string notr="true">1</string>
           </property>
          </column>
         </widget>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QLineEdit" name="scriptSearch">
          <property name="text">
           <string>Search...</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="scriptDelete">
          <property name="text">
           <string>Delete</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="scriptNew">
          <property name="text">
           <string>New</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2" rowspan="3">
         <widget class="QWidget" name="editContainer" native="true">
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QFrame" name="editControls">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QLabel" name="labelFilename">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="scriptPrevious">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="scriptNext">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="scriptSave">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="scriptClose">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QTextEdit" name="scriptBaseEdit">
             <property name="enabled">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="itemTab">
       <attribute name="title">
        <string>Item Editor</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="ItemEditor" name="itemEditor" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabDialogEditor">
       <attribute name="title">
        <string>Dialogue Editor</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="1" column="0" rowspan="2" colspan="5">
         <widget class="DialogEditor" name="dialogEditor" native="true">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QPushButton" name="dialogDelete">
          <property name="text">
           <string>Delete</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="3">
         <widget class="QComboBox" name="dialogList"/>
        </item>
        <item row="0" column="3">
         <widget class="QPushButton" name="dialogNew">
          <property name="text">
           <string>New</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Internationalization</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QPushButton" name="languageNew">
          <property name="text">
           <string>New</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="languageDelete">
          <property name="text">
           <string>Delete</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2" rowspan="2">
         <widget class="FormInternationalization" name="languageEditor" native="true">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QListWidget" name="languageList"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QPandaWidget</class>
   <extends>QWidget</extends>
   <header>qpandawidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>DialogEditor</class>
   <extends>QWidget</extends>
   <header>dialogeditor.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FormInternationalization</class>
   <extends>QWidget</extends>
   <header>forminternationalization.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ItemEditor</class>
   <extends>QWidget</extends>
   <header>itemeditor.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>WorldmapEditor</class>
   <extends>QWidget</extends>
   <header>worldmapeditor.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CharsheetEditor</class>
   <extends>QWidget</extends>
   <header>charsheeteditor.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>MapTreeWidget</class>
   <extends>QTreeWidget</extends>
   <header>maptreewidget.h</header>
  </customwidget>
  <customwidget>
   <class>WorldObjectWidget</class>
   <extends>QWidget</extends>
   <header>worldobjectwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
